### ISSUE TITLE
Acid armor doesn't appear to protect from ranged acid attacks

### ISSUE THREAD

**Issue by drhead** [CleverRaven/Cataclysm-DDA#55482]
### Describe the bug

When I am using armor that lists acid protection, I seem to take just as much damage from ranged acid attacks as I do with less acid-protective armor or no armor.

### Steps to reproduce

1. Spawn a combat exoskeleton and environmental combat helmet and wear them -- these have 7 acid armor and 100% full-body coverage should have no trouble blocking acid attacks that do about 3 damage.
2. Spawn a bilious soldier zombie.
3. Let the zombie use its acid dart gun attack, which deals a base of 3 acid damage.
4. Note that you take full damage from the attack.

### Expected behavior

I expect that armor that lists acid protection would protect me from acid attacks and somehow reduce or eliminate the damage.  Either acid protection is completely broken for attack damage purposes or some undocumented mechanic is allowing acid damage to bypass armor.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 97018ef [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Personal Blacklist [no_gas_zombies],
    Bionic Professions [package_bionic_professions],
    Aftershock [aftershock],
    Blaze Industries [blazeindustries],
    Crazy Cataclysm [crazy_cataclysm]
]

**Comment by drhead** [comment]
Lab coats (IRL) can protect you to an extent from chemical splashes and spills (though in game they provide zero acid protection, perhaps this should be fixed), but they probably wouldn't protect you from being drenched in acid, and a high-velocity acid projectile might not just roll off of it like a spill would.  I'd see that as a reasonable form of slight acid resistance.  Additionally, pretty much any cloth armor will absorb some amount of acid and prevent it from reaching your skin (though most likely at the expense of the armor itself -- but that fits in exactly with how armor durability works already, hits that go through the armor damage it, so it's fine).

Though with a combat exoskeleton, like in my example, the armor is made of heavy steel and kevlar plates, whatever "hyperweave" is, and a skintight, chemical resistant bodyglove.  If anything, acid attacks should mostly be polishing my armor.

**Comment by a-chancey** [comment]
I agree, it's odd that I can melee a group of hulks in my power armor without taking damage, and the one corrosive zombie that's behind the group is actively killing me by barfing acid on me - with zero ways to mitigate other than "kill it from a distance". It's somehow radiation proof, but acid gets right through and burns my torso for full damage. 

Since it specifically mentions the "chemical resistant bodyglove", I think either the acid protection should be high enough to protect you from acid barf, or the description should be changed, because it doesn't resist any of the chemicals that cause damage. It *is* waterproof, and with 100% coverage, should have zero chance of letting acid through based on the descriptions. 

And yes, this would buff power armor even more - acid barf is one of the very few things that threaten a power armor (especially a heavy combat exo) user currently. That list is: fire/heat, drowning, explosives, high caliber rifles and spells that drain stats or cause pain (like mi-go beams and flaming eyes), and being run over by your own vehicle.

**Comment by Night-Pryanik** [comment]
Looks similar to #55221. Seems like acid protection isn't working at all.
